tests/UnitTests/ParticipantManagementServicesTests/AddParticipantTest/AddParticipantTest.cs

```
@@ -173,6 +243,59 @@ public async Task Run_Should_Log_Participant_Marked_As_Eligible()
             ));
     }
     [TestMethod]
+public async Task Run_ParticipantEligibility_Failure()
```

There appears to be no assertion in this test. what is it doing?

---

it is also logging error, updated the test case name accordingly

---

updated the test cases with the respective assertions, unit test coverage now increased to 84%

---

your test name suggests that the test should only pass when the function logs an error, but your assertion is checking whether or not update eligibility has been called. Is the condition for the test passing that an error is logged or that the function is called?

---

this test cases is checking the negative scenario where if the HttpStatusCode is not OK then it should throw an exception. It is not checking the exception log message. If the test case name is creating a confusion , then I have changed the name to Run_ParticipantEligibilityFailure(), just confirm whether it works, or I'll update it to a new name.

---

Ok so you are testing if the function throws an exception when mark participant as eligible fails? In that case you need to name the test `Run_MarkAsEligibleFails_ThrowException`. However this is not what your test is doing. You are testing whether or not mark as eligible has been called, so you need to update the assertion to check if the function throws an exception.

---

updated the test method name and added the exception message assertion as asked.